wnfs/src/private/file.rs

```diff
@@ -554,36 +554,45 @@ impl PrivateFile {
     /// This should be called to prepare a node for modifications,
     /// if it's meant to be a successor revision of the current revision.
     ///
-    /// It will store the current revision in the given `BlockStore` to
-    /// retrieve its CID and put that into the `previous` links,
-    /// as well as advancing the ratchet and resetting the `persisted_as` pointer.
-    pub(crate) async fn prepare_next_revision(
-        self: Rc<Self>,
-        store: &mut impl BlockStore,
-        rng: &mut impl RngCore,
-    ) -> Result<Self> {
+    /// This doesn't have any effect if the current state hasn't been `.store()`ed yet.
+    /// Otherwise, it clones itself, stores its current CID in the previous links and
+    /// advances its ratchet.
+    pub(crate) fn prepare_next_revision(self: Rc<Self>) -> Result<Self> {
+        let previous_cid = match self.content.persisted_as.get() {
+            Some(cid) => *cid,
+            None => {
+                // The current revision wasn't written yet.
+                // There's no point in advancing the revision even further.
+                return Ok(Rc::try_unwrap(self).unwrap_or_else(|rc| (*rc).clone()));
```

Comment:
Notice we have this pattern in a couple places. Thinking we should bring `UnwrapOrClone` trait back.

Reply:
Yeah I know. The only reason I didn't bring it back was because we'll get rid of these as soon as we'll do the `&mut Rc<Directory>` stuff, since then the signature of this function should change to `(self: &mut Rc<Self>) -> &mut Self` and it makes use of `Rc::make_mut` instead.